### ISSUE TITLE
A3 is used twice in the software

### ISSUE THREAD

**Issue by zanco** [DO4AS/jufo-project-radiosonde#2]
Hi in config.h line 76 A3 is defined as solar_vin_pin

But in aprs.h line 28 A3 is defined as DIO2 pin.

I changed this setting in aprs.h to -1 and it appears to be still working without problems.

Can you confirm that DIO2 is not needed for the APRS transmission ?

Thanks,
Ben = PE2BZ

**Comment by DL9AS** [comment]
Hello Ben,

Thanks for posting this issue. In fact, A3 was defined twice. In the latest radiosonde hardware, A3 is actually used to measure solar input voltage.

On some older prototype radiosonde, I used a different library to interface the SX1278 radio chip. A3 was at that time connected to DIO2. DIO2 was defined in aprs.h. To ensure capability, I left this definition in the code. But to avoid further confusion and to clean up the code, I will remove this in the next commit. 



> Can you confirm that DIO2 is not needed for the APRS transmission ?

DIO2 is required in the code. After the migration to the current library, sx1278_radio_pwm_pin in config.h is connected to DIO2. DIO2 is used to generate the LF-modulation of the AFSK signal.

Hope this helps and greetings

Amon Schumann - DO4AS

BTW: Have you already tried to use this code on your hardware? What are your experiences? I would love to know.

**Comment by zanco** [comment]
Hi Amon, thanks for your explanation. 

Yes, I have built a prototype version, on a Pi Zero "hat" which had space for the GPS module and the SX1278 board and wired the necessary pins to an arduino Pro mini runing at 8 Mhz.

All text below is ment as positive comments ;-)

First conclusion: It works 👍  Somehow I managed to wire GPS output to Arduino TX but nothing happened thereafter.

After changing that I got serial output. That is something I would consider, start serial output by mentioning if either sensor is enabled, check i2c connection to the BME type sensor and print if there is GPS output detected. At this moment the serial output only starts after detecting GPS data which makes debugging a bit harder.

I also think that the GPS altitude is not transmitted if not either one of the sensors is enabled. With the BME enabled but not connected I do not get serial monitor output, so I disabled that one again and enabled the DS18b20. After that I got serial output and the tracker transmits a /Fkhdhfkdhfk telemetry like string and reports altitude on aprs software (and aprs.fi ) too.

The actual sensor temperatue I cannot find back however ?

I would have expected the telemetry string (or single altitude info) to be transmitted too if no sensor was enabled. 

Those experiences (except the mixup of GPS TX to arduino TX) are also confirmed by PC4L who built a prototype too.

Now going to study BME280 :-) 

Kind regards,
Ben
